File: cpp-main/ADS/lab 9/a.py
import math

def min_repetitions_to_contain(A, B):
    min_repeats = math.ceil(len(B) / len(A))
    
    repeated_A = A * min_repeats
    if B in repeated_A:
        return min_repeats
    elif B in repeated_A + A:
        return min_repeats + 1
    elif B in repeated_A + A * 2:
        return min_repeats + 2
    
    return -1

A = input().strip()
B = input().strip()
print(min_repetitions_to_contain(A, B))

File: cpp-main/ADS/lab 9/a.cpp
#include <iostream>
#include <string>

using namespace std;

class StringRepeater {
private:
    string A, B;

    bool isSubstring(const string& repeatedA) {
        return repeatedA.find(B) != string::npos;
    }

public:
    StringRepeater(const string& a, const string& b) : A(a), B(b) {}

    int minRepetitionsToContain() {
        int minRepeats = B.size() / A.size() + (B.size() % A.size() != 0 ? 1 : 0);
        string repeatedA = "";
        for(int i = 0; i < minRepeats; i++)
            repeatedA += A;
        
        if (isSubstring(repeatedA)) {
            return minRepeats;
        } else if (isSubstring(repeatedA + A)) {
            return minRepeats + 1;
        } else if (isSubstring(repeatedA + A + A)) {
            return minRepeats + 2;
        }
        
        return -1;
    }
};

int main() {
    string A, B;
    cin >> A >> B;

    StringRepeater repeater(A, B);
    int result = repeater.minRepetitionsToContain();

    cout << result << endl;
    
    return 0;
}
